datajunction-server/datajunction_server/api/notifications.py

```diff
@@ -0,0 +1,88 @@
+"""Dependency for notifications"""
+
+import logging
+from typing import Optional
+
+from fastapi import Depends
+from fastapi.responses import JSONResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from datajunction_server.database.history import ActivityType, EntityType, History
+from datajunction_server.database.notification_preference import NotificationPreference
+from datajunction_server.database.user import User
+from datajunction_server.internal.access.authentication.http import SecureAPIRouter
+from datajunction_server.utils import get_and_update_current_user, get_session
+
+router = SecureAPIRouter(tags=["notifications"])
+_logger = logging.getLogger(__name__)
+
+
+def get_notifier():
+    """Returns a method for sending notifications for an event"""
+
+    def notify(event: History):
+        """Send a notification for an event"""
+        _logger.debug("Sending notification for event %s", event)
+
+    return notify
+
+
+@router.post("/notifications/subscribe")
+async def subscribe_to_notifications(
```

Comment:
What about removing notifications? Maybe we should change this to `update_notifications` which would wipe them out if the args are empty or add a separate `delete_notifications` ?

Reply:
Oh right, how could I have forgotten that. It's like when you can subscribe to a server with 2 clicks on your phone but if you want to cancel they route through 43 different call centers and then make you mail them a handwritten letter.